app/models/user.rb

```diff
@@ -46,6 +48,15 @@ class User < ActiveRecord::Base
   # We really need a Dispatch Chain here or something.
   # This will also let us return a human error message.
   #
+
+  def self.from_omniauth(auth)
```

Comment:
This method need specs. It will be nice to test both the find and the creation paths.